### ISSUE TITLE
KeyError when attempting to upload an translation file

### ISSUE THREAD

**Issue by superDross** [WeblateOrg/weblate#4245]
**Describe the bug**

When trying to [upload a PO file](https://docs.weblate.org/en/latest/user/files.html#uploading-translations) to overwrite existing translations I am met with an KeyError referring to `author_name`.

I suspect this is because the upload file form should have an author name and author email field but does not.

**To Reproduce**

Via the admin interface:
- Create a role with every single permission 
- Create a permission group with newly created role
- Assign group to desired user

Via the Weblate instance:
- Go to the upload translation file as described [here](https://docs.weblate.org/en/latest/user/files.html#uploading-translations) 
- Uploading a file and picking any option will trigger it but I am interested in `Replace exisintg translation file` and `Import as translated`
- Press the `Upload` button

**Expected behavior**

A message telling me the outcome of uploading the translation file.

**Server configuration and status**

 * Weblate: 4.1.1
 * Django: 3.0.8
 * siphashc: 1.3
 * Whoosh: 2.7.4
 * translate-toolkit: 3.0.0
 * lxml: 4.5.1
 * Pillow: 7.1.2
 * bleach: 3.1.5
 * python-dateutil: 2.8.1
 * social-auth-core: 3.3.3
 * social-auth-app-django: 3.4.0
 * django-crispy-forms: 1.9.1
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.11.0
 * django-filter: 2.3.0
 * django-appconf: 1.0.4
 * user-agents: 2.1
 * filelock: 3.0.12
 * setuptools: 40.8.0
 * jellyfish: 0.8.2
 * openpyxl: 3.0.1
 * celery: 4.4.6
 * kombu: 4.6.11
 * translation-finder: 2.1
 * html2text: 2020.1.16
 * pycairo: 1.16.2
 * pygobject: 3.30.4
 * diff-match-patch: 20181111
 * requests: 2.24.0
 * django-redis: 4.12.1
 * hiredis: 1.0.1
 * sentry_sdk: 0.15.1
 * Cython: 0.29.20
 * misaka: 2.1.1
 * GitPython: 3.1.3
 * borgbackup: 1.1.13
 * pyparsing: 2.4.7
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2: 2.8.5
 * psycopg2-binary: 2.8.5
 * phply: 1.2.5
 * chardet: 3.0.4
 * ruamel.yaml: 0.16.10
 * tesserocr: 2.5.1
 * akismet: 1.1
 * boto3: 1.14.14
 * zeep: 3.4.0
 * aeidon: 1.7.0
 * iniparse: 0.5
 * mysqlclient: 1.4.6
 * Mercurial: 5.4.2
 * git-svn: 2.20.1
 * git-review: 1.28.0
 * hub: 2.13.0
 * lab: 0.16
 * Redis server: 5.0.9
 * PostgreSQL server: 11.8
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: smtp.mailtrap.io
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 5.4.52-1-MANJARO (x86_64)

**Exception traceback**

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/dist-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/files.py", line 146, in upload_translation
    author_name = form.cleaned_data["author_name"]

Exception Type: KeyError at /upload/bilingual-vs-monolingual/monolingual/aa/
Exception Value: 'author_name'
```
**Screen Shot**
![thiingy](https://user-images.githubusercontent.com/16519378/89189320-b9128c80-d597-11ea-8972-29118720b6f8.png)

As can be seen in the above screen shot, no author name or author email fields are present as one would expect.

**Additional context**

This works fine when you add the `<component-name>@Administration` group to the users chosen groups via the admin interface. The difference being the Author name and email fields are available in the upload translation form.

**Comment by nijel** [comment]
Any component level permission in place here? In case it is, it will be most likely fixed by ab0a6a1fb2af24085db372db09ed0c0bf27dbde8.

**Comment by superDross** [comment]
Yeah there is. Once I deselect all components from the permissions group I can see the author name and author email fields available in the upload translation form. 

Thanks for the speedy reply.
